cirq/google/serializable_device_test.py

```diff
@@ -264,4 +265,52 @@ def test_mixing_types():
     gate.valid_targets.extend(['subset', 'sym'])
 
     with pytest.raises(NotImplementedError):
-        _ = cg.SerializableDevice.from_proto(proto=spec, gate_set=_JUST_MEAS)
+        _ = cg.SerializableDevice.from_proto(proto=spec, gate_sets=[_JUST_MEAS])
+
+
+def test_multiple_gatesets():
+    halfPiGateSet = cirq.google.serializable_gate_set.SerializableGateSet(
+        gate_set_name='half_pi_gateset',
+        serializers=cgc.SINGLE_QUBIT_HALF_PI_SERIALIZERS,
+        deserializers=cgc.SINGLE_QUBIT_HALF_PI_DESERIALIZERS,
+    )
+    allAnglesGateSet = cirq.google.serializable_gate_set.SerializableGateSet(
+        gate_set_name='half_pi_gateset',
+        serializers=cgc.SINGLE_QUBIT_SERIALIZERS,
+        deserializers=cgc.SINGLE_QUBIT_DESERIALIZERS,
+    )
+    durations_dict = {'xy_pi': 20_000, 'xy_half_pi': 20_000, 'xy': 20_000}
+    spec = cirq.google.known_devices.create_device_proto_from_diagram(
+        "aa\naa", [allAnglesGateSet, halfPiGateSet], durations_dict)
+    dev = cg.SerializableDevice.from_proto(
+        proto=spec, gate_sets=[allAnglesGateSet, halfPiGateSet])
+    dev.validate_operation(cirq.X(cirq.GridQubit(0, 0)))
+    dev.validate_operation(cirq.X(cirq.GridQubit(1, 0)))
+
+    with pytest.raises(ValueError):
+        dev.validate_operation(cirq.X(cirq.GridQubit(2, 2)))
+
+
+def test_multiple_gatesets_conflicting_definitions():
+    halfPiGateSet = cirq.google.serializable_gate_set.SerializableGateSet(
+        gate_set_name='half_pi_gateset',
+        serializers=[
+            *cgc.SINGLE_QUBIT_HALF_PI_SERIALIZERS, cgc.MEASUREMENT_SERIALIZER
+        ],
+        deserializers=[
+            *cgc.SINGLE_QUBIT_HALF_PI_DESERIALIZERS,
+            cgc.MEASUREMENT_DESERIALIZER
+        ],
+    )
+    durations_dict = {
+        'xy_pi': 20_000,
+        'xy_half_pi': 10_000,
+        'exp_w': 53_000,
+        'exp_11': 11_000,
+        'meas': 14_141
+    }
+    spec = cirq.google.known_devices.create_device_proto_from_diagram(
+        "aa\naa", [cirq.google.gate_sets.XMON, halfPiGateSet], durations_dict)
+    with pytest.raises(ValueError):
+        _ = cg.SerializableDevice.from_proto(
+            proto=spec, gate_sets=[_JUST_MEAS, halfPiGateSet])
```

Comment:
It's not exactly clear what is the conflict here. It looks like the gate_sets will be missing the XMON gates, but that would seem to be a different error condition from conflicting definitions as implied by the test name.

Reply:
Changed this test around to make it more readable and more robust.

Also, it was depending on XMON and SINGLE_QUBIT to serialize into different symbols, hence the conflict, but I am removing that distinction in a different PR.